### ISSUE TITLE
Versioning Google vs F-Droid

### ISSUE THREAD

**Issue by pm-pm** [stefan-niedermann/nextcloud-notes#680]
Hello,

i have problems to understand the version information from Google vs. F-Droid.
Is there a mapping list or something else? Are there different Features?

Thanks.

**Comment by stefan-niedermann** [comment]
They are absolutely identical, the versions from Play Store are build from the [Git-tags](https://github.com/stefan-niedermann/nextcloud-notes/releases).

The only difference is, that F-Droid builds new tägs instantly and delivers the new versions, while in Play Store, i do staged releases (for example: Day 0: 10%, Day 5: 20%, Day 21: 100%).

The reason is simply that if something goes wrong, it is possible quickly cancel the release, build a new bug fix release, and many people will never know there was a bug.

Current Play Store situation is like ~85% are on version `1.0.x`, 10% are on a `2.x.x`-version.

The new Single Sign On introduced in version `2.0.0` still does not work 100% correctly for some edge cases, and i stopped further updates ät Play Store for now to investigate and fix the remaining issues. Äs soon as most harder issues are resolved, the fixed version will be rolled out, then.

**TL;DR: Pläy Store does not receive updates that often but only version for all users which are known to have no bugs left.**

---

If you want to receive more updates (nearly each tagged version) you can sign up for the beta channel: https://play.google.com/apps/testing/it.niedermann.owncloud.notes

**Comment by pm-pm** [comment]
Hello Stefan,

thanks for detailed reply. :-) I am still confused.

> Current Play Store situation is like ~85% are on version `1.0.x`, 10% are on a `2.x.x`-version.

There is a "2.x.x" Version on Google? I can only see 1.0.1.

> **TL;DR: Play Store does not receive updates that often but only version for all users which are known to have no bugs left.**

That is a good idea! But a very small mapping list (maybe only for bug-free-releases) between Google and F-Droid would be helpful

Does anything match to each other?
Google only : 1.0.1
F-Droid only: 2.40, 2.41, 2.44

Is it possible to get a old? / bugfree? 1.0.1 Version on F-Droid.

> If you want to receive more updates (nearly each tagged version) you can sign up for the beta channel: https://play.google.com/apps/testing/it.niedermann.owncloud.notes

No, i can not. I have not G**gle Account and feel good. ;-)

Thanks a lot for your work and help!

pm-pm

**Comment by stefan-niedermann** [comment]
**Each version maps exactly to the correspondenting F-Droid-Version.**

1.0.1 on Play Store is exactly the same like 1.0.1 on F-Droid
2.4.6 on Play Store is exactly the same like 2.4.6 on F-Droid

and so on.

Play Store has both versions, but the newer version is only visible for a small group of people. This group continously increases until all people have the new version. This concept is called "[staged rollout](https://medium.com/bleeding-edge/the-art-of-staging-a-rollout-8e203b337b75)"

You can check the new features and bug fixes [here](https://github.com/stefan-niedermann/nextcloud-notes/tags). Not every version is released at Play Store (e.g. if it contains known bugs or if the change is only minimal), so i group the release notes for Play Store from the last to the current release.

**Comment by stefan-niedermann** [comment]
> Is it possible to get a old? / bugfree? 1.0.1 Version on F-Droid.

Sure, by default F-Droid lists only the last three versions for performance reasons, but all versions are available: just go to your F-Droid options and enable the "Archive" repository in your package sources.

What do you mean with "bugfree"? If you got any issues with the current version, it's always better to report it here instead of silently stuck on an old version.

**Comment by pm-pm** [comment]
Hello Stefan, 

MANY thanks for the information.

"Bugfree" means what you said above. :-)

>  "only version for all users which are known to have no bugs left"

News bugs i would report here of course.


Now i really understand the Versioning. It is only so confusion, because the "numbers" between the two plattforms are so wide apart. "1.0.1" vs. "2.44" But with your background information i understand it very well.

> enable the "Archive" repository 

This must be a feature of the f-droid apk. On the f-droid.org Webpage i can not find this feature.

So you maybe understand my "problems" if you only compare the two webpages. Everything is now clear to me.

Maybe you add the information from your last two postings also here:
> https://github.com/stefan-niedermann/nextcloud-notes

This could be also helpful for other users.

Thanks a lot.

pm-pm

**Comment by stefan-niedermann** [comment]
You're welcome :)

The current situation is special because i increased the major version **and** i cannot quickly rollout the 2.x.x version on Play Store completely.

I'll consider adding it to the README.md, but i doubt that many users will find this helpful.
